spec/unit/methods/off.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { FramebusConfig } from "../../../src/framebus-config";
+import { subscribers } from "../../../src/lib/constants";
+import { off, on, teardown } from "../../../src/methods";
+
+describe("off", () => {
+  afterEach(() => {
+    Object.keys(subscribers).forEach((subKey) => {
+      delete subscribers[subKey];
+    });
+  });
+
+  describe("with origin", () => {
+    let config: FramebusConfig;
+    const origin = "https://example.com";
+    beforeEach(() => {
+      config = new FramebusConfig({
+        origin,
+      });
+    });
+
+    it("should remove subscriber given event and origin", () => {
+      const event = "the event";
+      const fn = jest.fn();
+
+      subscribers[origin] = {};
+      subscribers[origin][event] = [jest.fn(), fn];
+
+      off(config, event, fn);
+
+      expect(subscribers[origin][event]).not.toContain(fn);
+      expect(subscribers[origin][event].length).toBe(1);
+    });
+
+    it("should correctly update the array", () => {
+      const event = "the event";
+      const fn = jest.fn();
+
+      subscribers[origin] = {};
+      subscribers[origin][event] = [jest.fn(), fn];
+
+      off(config, event, fn);
+
+      expect(subscribers[origin][event].length).toBe(1);
```

Comment:
This test assertion is also made in the test above. Could potentially consolidate or have a single assertion per test.